### PR TITLE
Switch to building GMP from an unofficial git mirror (CI hash update)

### PR DESCRIPTION
**Pull request by cameel**
Depends on #14767 (should be merged together).

~**Note**: I moved the ossfuzz job to the main workflow to test the image. This will be removed before the PR is merged.~ Done.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1860,10 +1864,6 @@ workflows:
           <<: *on_develop
 
     jobs:
-      # OSSFUZZ builds and (regression) tests
-      - b_ubu_ossfuzz: *requires_nothing
```
Comment (bshastry) [inline .circleci/config.yml]: Why is this line removed?

**Review by bshastry — CHANGES_REQUESTED** [review]
If CI passes, fuzz test workflow needs to be moved back to nightly. Otherwise have one question?

File: .circleci/config.yml (hunk 1 above)
Comment (cameel) [reply .circleci/config.yml]: Just for testing. I'll restore it in a moment.

I mean, I was getting and error because of a duplicate `b_ubu_ossfuzz` in the workflow. Later I noticed that this is not because of this but another instance of it that's at the top of the `main` workflow. But it does not matter in this PR. `nightly` won't run here anyway.

**Comment by cameel** [comment]
Looks like CI kinda passed. I.e. [`t_ubu_ossfuzz` did fail like it does in `nightly`](https://app.circleci.com/pipelines/github/ethereum/solidity/32268/workflows/c8966e62-37b8-4e83-8709-b0a3a8d5e526/jobs/1442994), but not because of anything GMP-related so I'm going to assume it's fine.

And brink failure is unrelated - standard ext test breakage. We can fix it separately.

So I'm removing the debug commit and making this reviewable.

**Review by bshastry — DISMISSED** [review]
Lgtm
